cmd/data-node/commands/init.go

```diff
@@ -31,10 +32,16 @@ type InitCmd struct {
 
 var initCmd InitCmd
 
-func (opts *InitCmd) Execute(_ []string) error {
+func (opts *InitCmd) Execute(args []string) error {
 	logger := logging.NewLoggerFromConfig(logging.NewDefaultConfig())
 	defer logger.AtExit()
 
+	if len(args) != 1 {
```

Comment:
Do we really need to provide this during init, and not just left for the user to fill in the config file afterwards (or they can do `datanode dehistory load --chainID=BLAH`? Feels odd that it needs to be supplied at init like its tied to this initialisation of the datanode, but then you can just change it at any time in the config?

Reply:
the datanode uses the chain id to connect to the correct swarm at startup, such that when the user does `data-node node` it can automatically determine its peer set, get the latest history segment, load it, and start.  Without this the alternative would mean either passing it in on `data-node node` each time, or having the user do a seperate process to load the data.  In order to make the experience of starting up a new node as seemless as possible we want to avoid the latter and have it basically get its self into the correct state to be able to process the latest blocks without any intervention.

Reply:
Sure, but its in a config file so I can change it at any point right? What happens if I do that, stop the data-node then start it again with a different chainID? I hoping something with tell me I'm being silly.

I just think anything that gets set at the `init` call should only be able to be changed if I reinitialise with `init --force`.

We also have a check in the broker that makes sure that the chainID sent by the core node never changes by saving the first chainID it receives, and thats saved into the database (I think):
```
func checkChainID(expectedChainInfo ChainInfoI, chainID string) error {}
```
I think this now probably needs to be extended to make sure that the core node we're connected to is from the same chainID we initialised the datanode to be with.

Reply:
rgds the broker it already has a check for that, if data is loaded from decentralized history and then the node starts to consume data from a different chain it will error as loading decentralized history sets the chain id that is used by the broker to verify the events chain.   I agree it would be preferable to use the chain set in config for this check as well though, I'll make that change.

It doesn't explicitly check the chain id in config matches that in data at startup at the moment, it probably should this will prevent any unexpected behaviour, I'll add that.

The main thing here is are we happy with it being passed at init?  I can think of other ways of doing this, but they seem a bit brittle and over-complicated and make it harder to start a data-node in isolation, I purposely raised this at the demo and there was silence, which was sort of agreement I guess ;-)

Reply:
Haha sure keep it at during `init`, you're right that no one else complained at the demo.